### PR TITLE
gh-94801: Use custom flags to check for headers and libs

### PR DESCRIPTION
**Pull request by tiran**
* Issue: gh-94801

Automerge-Triggered-By: GH:tiran

### CODE REVIEW

**Comment by tiran** [comment]
I addressed most places. The code now correctly triggers Emscripten's embuilder to download and install ports.

File: configure.ac [hunk 1]
```diff
@@ -3371,12 +3379,12 @@ AS_VAR_IF([have_uuid], [missing], [
...
       AC_DEFINE([HAVE_UUID_GENERATE_TIME_SAFE], [1])
     ], [
-      AC_CHECK_HEADERS([uuid/uuid.h], [
-        WITH_SAVE_ENV(
-          [AC_CHECK_LIB([uuid], [uuid_generate_time], [have_uuid=yes])
-        ])
-        WITH_SAVE_ENV([
-          AC_CHECK_LIB([uuid], [uuid_generate_time_safe], [
+      WITH_SAVE_ENV([
+        CFLAGS="$CFLAGS $LIBUUID_CFLAGS"
```
Comment (erlend-aasland) [inline configure.ac]: Would it make sense to consistently use CPPFLAGS for these checks? IIRC, configure on FreeBSD had some issues with CFLAGS in one of my sqlite3 PRs.

File: configure.ac [hunk 2]
```diff
@@ -3779,6 +3790,12 @@ AS_CASE([$ac_sys_system],
   [OSSAUDIODEV_LIBS=""]
 )
 
+dnl detect sqlite3 from Emscripten emport
+if test "$ac_sys_system" = "Emscripten" -a -z "$LIBSQLITE3_CFLAGS" -a -z "$LIBSQLITE3_LIBS"; then
+  LIBSQLITE3_CFLAGS="-sUSE_SQLITE3"
+  LIBSQLITE3_LIBS="-sUSE_SQLITE3"
+fi
```
Comment (erlend-aasland) [inline configure.ac]: We _could_ make a macro for these Emscripten checks 😎

<details>
<summary>PoC</summary>

```diff
diff --git a/configure.ac b/configure.ac
index 43c9802870..05de6239be 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3790,11 +3790,20 @@ AS_CASE([$ac_sys_system],
   [OSSAUDIODEV_LIBS=""]
 )
 
+dnl PY_CHECK_EMPORT_SUPPORT(LIB)
+AC_DEFUN([PY_CHECK_EMPORT_SUPPORT], [
+  AS_VAR_PUSHDEF([py_cflags], [LIB$1_CFLAGS])
+  AS_VAR_PUSHDEF([py_libs], [LIB$1_LIBS])
+  AS_IF([test "$ac_sys_system" = "Emscripten" -a -z "$py_cflags" -a -z "$py_libs"], [
+    py_cflags="-sUSE_$1"
+    py_libs="-sUSE_$1"
+  ])
+  AS_VAR_POPDEF([py_cflags])
+  AS_VAR_POPDEF([py_libs])
+])
+
 dnl detect sqlite3 from Emscripten emport
-if test "$ac_sys_system" = "Emscripten" -a -z "$LIBSQLITE3_CFLAGS" -a -z "$LIBSQLITE3_LIBS"; then
-  LIBSQLITE3_CFLAGS="-sUSE_SQLITE3"
-  LIBSQLITE3_LIBS="-sUSE_SQLITE3"
-fi
+PY_CHECK_EMPORT_SUPPORT([SQLITE3])
 
 dnl Check for SQLite library. Use pkg-config if available.
 PKG_CHECK_MODULES(
@@ -4806,10 +4815,7 @@ dnl Check for compression libraries
 AH_TEMPLATE([HAVE_ZLIB_COPY], [Define if the zlib library has inflateCopy])
 
 dnl detect zlib from Emscripten emport
-if test "$ac_sys_system" = "Emscripten" -a -z "$ZLIB_CFLAGS" -a -z "$ZLIB_LIBS"; then
-  ZLIB_CFLAGS="-sUSE_ZLIB"
-  ZLIB_LIBS="-sUSE_ZLIB"
-fi
+PY_CHECK_EMPORT_SUPPORT([ZLIB])
 
 PKG_CHECK_MODULES([ZLIB], [zlib >= 1.2.0], [
   have_zlib=yes
@@ -4837,10 +4843,7 @@ AS_VAR_IF([have_zlib], [yes], [
 ])
 
 dnl detect bzip2 from Emscripten emport
-if test "$ac_sys_system" = "Emscripten" -a -z "$BZIP2_CFLAGS" -a -z "$BZIP2_LIBS"; then
-  BZIP2_CFLAGS="-sUSE_BZIP2"
-  BZIP2_LIBS="-sUSE_BZIP2"
-fi
+PY_CHECK_EMPORT_SUPPORT([BZIP2])
 
 PKG_CHECK_MODULES([BZIP2], [bzip2], [have_bzip2=yes], [
   WITH_SAVE_ENV([

```

</details>

**Review by erlend-aasland — APPROVED** [review]
Nice! LGTM.

File: configure.ac (hunk 2 above)
Comment (tiran) [reply configure.ac]: I modified your approach a bit. There is no direct 1:1 mapping between pkg-config and emports args.

File: configure.ac (hunk 1 above)
Comment (tiran) [reply configure.ac]: done

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**